### DISCUSSION TITLE
EndpointPickers example works on macOS but not on iOS

### DISCUSSION THREAD

**Discussion by kempline** [orchetect/swift-midi#201] [Q&A] [answered]
Hi all, great project!

I tried compiling some of the examples and it went pretty well!

After compiling, I connected a guitar preamp via USB to my macOS device. The example came up and I was able to pick source and destination. Since the preamp only accepts program changes, I added some code in both files, ContentView-iOS and  ContentView-macOS:

```
Button("Send PC1") {
    sendToConnection(.programChange(program: 1, channel: 0x3))
}

Button("Send PC2") {
    sendToConnection(.programChange(program: 2, channel: 0x3))
}
```

On macOS, the messages get fired and the preamp switches from program 2 to 3 (counting starts at 0 in the preamp). Running the same code on my iPhone, the iOS device doesn't seem to fire anything. To make sure that there is no hardware issue, I tried sending PC messages to the same preamp using another software called Bandhelper. As expected, this works well.

Since it works on macOS, I assume I have to set-up something in the iOS device to make the messages going through?

Thanks for ideas!

**Comment by NickCulbertson** [comment]
@kempline This should fix your issue. https://github.com/orchetect/MIDIKit/pull/202

**Reply by orchetect** [reply]
Thanks Nick - I will have a look and merge as soon as I can. Looks like it was just an oversight during UI refactors.

**Reply by kempline** [reply]
@NickCulbertson Thank you. Yes, now it works as expected.

**Reply by orchetect** [reply, marked as answer]
PR is merged. Fix will be in the next release, but until then you can pull main branch.
